logger/sdkLogger.go

```diff
@@ -0,0 +1,336 @@
+package logger
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+// SdkLoggerInterface Represents an interface for logging API requests and responses.
+type SdkLoggerInterface interface {
+	// LogRequest logs the details of an HTTP request.
+	LogRequest(request *http.Request)
+	// LogResponse logs the details of an HTTP response.
+	LogResponse(response *http.Response)
+}
+
+// SdkLogger represents implementation for SdkLoggerInterface, providing methods to log HTTP requests and responses.
+type SdkLogger struct {
+	loggingOptions LoggerConfiguration
+	logger         LoggerInterface
+}
+
+// NewSdkLogger Constructs a new instance of SdkLogger.
+func NewSdkLogger(loggingOpt LoggerConfiguration) *SdkLogger {
+	return &SdkLogger{
+		loggingOptions: loggingOpt,
+		logger:         loggingOpt.logger,
+	}
+}
+
+// LogRequest request Logs an HTTP request.
+func (a *SdkLogger) LogRequest(request *http.Request) {
+	var logLevel = a.loggingOptions.level
+	var contentTypeHeader = a._getContentType(request.Header)
+	var url string
+	if a.loggingOptions.request.includeQueryInPath {
+		url = request.RequestURI
+	} else {
+		url = a._removeQueryParams(request.RequestURI)
+	}
+
+	a.logger.Log(
+		logLevel,
+		fmt.Sprintf("Request %v %v %v", request.Method, url, contentTypeHeader),
+		map[string]any{
```

Comment:
@haseeb-mhr did Mujahid bhai went over this logging approach?

The purpose of structured logging is giving user a message format and parameters that they can replace in their message to get the real message.

Whereas you are replacing the message placeholders yourself before emitting the message to the loggerInterface^

Reply:
Updated this logic by implementing custom placeholder replacement.